app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx

```diff
@@ -1,23 +1,35 @@
 import React, { useState } from "react";
 
 import { Formik, Form } from "formik";
+import { Check } from "neetoicons";
 import { Button, Pane } from "neetoui";
-import { Input, Textarea } from "neetoui/formik";
+import { Input, Select } from "neetoui/formik";
 
-import notesApi from "apis/notes";
+import { CONTACTS, TAGS } from "components/Dashboard/constants";
+import { buildSelectOption } from "utils/index";
 
 import { NOTES_FORM_VALIDATION_SCHEMA } from "../constants";
 
-const NoteForm = ({ onClose, note, isEdit }) => {
+const tags = TAGS.map(buildSelectOption);
+const contacts = CONTACTS.map(contact =>
+  buildSelectOption(`${contact.firstName} ${contact.lastName}`)
+);
+
+const parseValues = values => {
+  const assignedContact = values.assignedContact.label;
+  const tags = values.tags.map(tag => tag.label);
+
+  return { ...values, tags, assignedContact };
+};
```

Comment:
1) Why are these literal constants named in camelCase?
2) Why are these utils kept in the component and not moved out into utils file?

Reply:
1. There weren't any hardcoded strings in that line, hence I felt it didn't qualify to be named so. But I will change it to SNAKE_CASE since you mentioned it.
2. That function had only been used in that particular file, with no use case in any other place even in the future, hence I kept it there itself. But I'll move it to the utils file anyway.👍

Reply:
>There weren't any hardcoded strings in that line, hence I felt it didn't qualify to be named so. But I will change it to SNAKE_CASE since you mentioned it.

That's not the right way to make a judgement on whether the value is a literal constant or not. Here `TAGS` is a literal constant. The logic for `buildSelectOption` is also pure, meaning it won't change in any circumstances. Thus `TAGS.map(buildSelectOption);` is also going to provide a literal constant with same value, no matter how many times we rerun the function.

>That function had only been used in that particular file, with no use case in any other place even in the future, hence I kept it there itself. But I'll move it to the utils file anyway.

I understand. But it's about keeping the presentation logic as clean as possible. Thus the suggestion.